src/hooks/SocialAuth.ts

```diff
@@ -0,0 +1,104 @@
+import { randomBytes } from 'crypto'
+import useLocalStorage from 'hooks/LocalStorage'
+import usePopupWindow from 'hooks/PopupWindow'
+import { useEffect, useState } from 'react'
+
+type OAuthData = {
+  code: string
+}
+
+type OauthOptions = {
+  client_id: string
+  scope: string
+  response_type?: 'code' | 'token'
+  code_challenge?: 'challenge'
+  code_challenge_method?: 'plain'
+}
+
+const useOauthPopupWindow = (url: string, oauthOptions: OauthOptions) => {
+  const [csrfToken, setCsrfToken] = useLocalStorage(
+    `oauth_csrf_token_${oauthOptions.client_id}`,
+    randomBytes(16).toString('hex'),
+    true,
+  )
```

Comment:
**potential issue**: if we're generating the csrf token on the client, then isn't this vulnerable to xss? Because anyone can set the csrf token to whatever they want.

Also, is the csrf token actually being checked anywhere?

Reply:
yeah, tbf a lot of this Oauth flow is copied from guild.xyz

https://github.com/agoraxyz/guild.xyz

the CSRF token is checked in the callback iirc, but that code is commented out and i forgot about it. i don't have the best understanding of this stuff so will look into it more